### PR TITLE
Standardization of expression parser inside parsinho

### PR DESCRIPTION
**Pull request by leonabdalla**
This PR is updating all expression in order to allow ExpressionDataParser to parse Expression in the same way LogicalPlanParser parse LogicalPlan.

### CODE REVIEW

File: src/main/scala/nubank/parsinho/expressions/ExpressionData.scala [hunk 1]
```diff
@@ -8,7 +9,9 @@ trait ExpressionData {
   def children: Seq[ExpressionData]
 }
 
-final case class NonParsedExpression(planName: String, dataType: DataType, children: Seq[ExpressionData])
+final case class NonParsedExpression(expression: Expression, dataType: DataType, children: Seq[ExpressionData])
```
Comment (jpmicena) [inline src/main/scala/nubank/parsinho/expressions/ExpressionData.scala:12]: I think you should also remove the `dataType` from the constructor and put it inside the case class body

File: src/main/scala/nubank/parsinho/expressions/unary/UnaryExpressionData.scala [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package nubank.parsinho.expressions.unary
 
 import nubank.parsinho.expressions.ExpressionData
+import org.apache.spark.sql.catalyst.expressions.ExprId
```
Comment (jpmicena) [inline src/main/scala/nubank/parsinho/expressions/unary/UnaryExpressionData.scala:4]: I think this is an unnecessary import (we need tests for that, that's something I want to explore later)

File: src/main/scala/nubank/parsinho/nodes/binary/JoinData.scala [hunk 3]
```diff
@@ -28,9 +29,10 @@ final case class JoinData(plan: Join, override val left: NodeData, override val
 
   // TODO: solve the existence, natural and using joins so we never have a None on `condition`
   // That way we can drop this condition value
+  @SuppressWarnings(Array("org.wartremover.warts.AsInstanceOf"))
   def conditionValue: ExpressionData = condition match {
     case Some(cond) => cond
-    case _          => NonParsedExpression("non parsed condition", BooleanType, Seq.empty[ExpressionData])
+    case _          => NonParsedExpression(Nil.asInstanceOf[Expression], BooleanType, Seq.empty[ExpressionData])
```
Comment (jpmicena) [inline src/main/scala/nubank/parsinho/nodes/binary/JoinData.scala:35]: hmmm I really don't like this `asInstanceOf` here, but again, this should be removed altogether so its cool for now :P

File: src/main/scala/nubank/parsinho/parser/ExpressionDataParser.scala [hunk 4]
```diff
@@ -12,25 +12,22 @@ object ExpressionDataParser {
...
+      case expr: And             => AndData(expr, fromExpression(expr.left), fromExpression(expr.right))
+      case expr: Or              => OrData(expr, fromExpression(expr.left), fromExpression(expr.right))
 
       // Catch-all for unimplemented expressions
-      case ouch =>
-        NonParsedExpression(ouch.getClass.getSimpleName,
-                            dataType = NullType,
-                            children = ouch.children.map(fromExpression))
+      case expr =>
+        NonParsedExpression(expr, dataType = NullType, children = expr.children.map(fromExpression))
```
Comment (jpmicena) [inline src/main/scala/nubank/parsinho/parser/ExpressionDataParser.scala:30]: Lindo e elegante 👏

File: src/main/scala/nubank/parsinho/expressions/ExpressionData.scala (hunk 1 above)
Comment (EEmery) [reply src/main/scala/nubank/parsinho/expressions/ExpressionData.scala:12]: I'm with @jpmicena, I think that having `dataType` inside the body is more consistent with the other `ExpressionData`s

File: src/main/scala/nubank/parsinho/nodes/binary/JoinData.scala (hunk 3 above)
Comment (EEmery) [reply src/main/scala/nubank/parsinho/nodes/binary/JoinData.scala:35]: Maybe an approach like #23 could be useful here

File: src/main/scala/nubank/parsinho/expressions/ExpressionData.scala (hunk 1 above)
Comment (leonabdalla) [reply src/main/scala/nubank/parsinho/expressions/ExpressionData.scala:12]: Ok done.

File: src/main/scala/nubank/parsinho/nodes/binary/JoinData.scala (hunk 3 above)
Comment (leonabdalla) [reply src/main/scala/nubank/parsinho/nodes/binary/JoinData.scala:35]: @EEmery will solve in his PR

File: src/main/scala/nubank/parsinho/expressions/unary/UnaryExpressionData.scala (hunk 2 above)
Comment (leonabdalla) [reply src/main/scala/nubank/parsinho/expressions/unary/UnaryExpressionData.scala:4]: you are right

**Pull request merged**